File: src/main/webapp/script/endoService.js
endoMinion.factory('endoService', function($http, $rootScope) {
	var service = this;
	return {
		getSimpleData : function() {
			return $http.get('/EndoMinion/getSimpleData');
		},
		getWorkouts : function() {
			return $http.get('/EndoMinion/getWorkouts');
		},
		getSports : function() {
			return $http.get('/EndoMinion/getSports');
		},
		fetchWorkouts : function() {
			return $http.get('/EndoMinion/fetchWorkouts');
		},
		getStatistics : function() {
			return $http.get('/EndoMinion/getStatistics');
		},
		getAccountInfo : function() {
			return $http.get('/EndoMinion/getAccountInfo');
		},
		prefetchData : function() {
			return $http.get('/EndoMinion/prefetchData');
		},
		getDetailedWorkout : function(endomondoId) {
			return $http.get('/EndoMinion/getDetailedWorkout/'+endomondoId);
		}
	}
});

File: src/main/java/org/gvaireth/endominion/server/EndoService.java
package org.gvaireth.endominion.server;

import java.util.List;

import org.gvaireth.endominion.model.AccountInfoData;
import org.gvaireth.endominion.model.DetailedWorkoutData;
import org.gvaireth.endominion.model.PrefetchDataStatus;
import org.gvaireth.endominion.model.StatisticsData;
import org.gvaireth.endominion.model.WorkoutData;

public interface EndoService {

	List<WorkoutData> getWorkouts();

	DetailedWorkoutData getWorkoutDetails(long endomondoId);

	List<WorkoutData> fetchWorkouts();

	StatisticsData getStatistics();

	AccountInfoData getAccountInfo();

	PrefetchDataStatus prefetchData();

	List<String> getSports();

}
